src/indicators/drought/ecmwf_seasonal.py

```diff
@@ -0,0 +1,301 @@
+        logger.debug(f"Querying for {filepath}...")
+        logger.debug(f"{self._get_query(area=area,version=version,year=year,month=month,leadtime=leadtime, )}")
+        CDSAPI_CLIENT.retrieve(
+            name=self.cds_name,
+            request=self._get_query(
+                area=area,
+                version=version,
+                year=year,
+                month=month,
+                leadtime=leadtime,
+            ),
+            target=filepath,
+        )
+        logger.debug(f"...successfully downloaded {filepath}")
+        # Wait 2 seconds between requests or else API hangs
+        # TODO make sure this actually works
+        time.sleep(2)
+        return filepath
+
+    def _get_raw_filepath(
+        self,
+        country_name: str,
+        country_iso3: str,
+        version: int,
+        year: int,
+        month: int = None,
+        leadtime: int = None,
+    ):
+        directory = (
+            RAW_DATA_DIR
+            / country_name
+            / ECMWF_SEASONAL_DIR
+            / self.cds_name
+        )
+        filename = f"{country_iso3}_{self.cds_name}_v{version}_{year}"
+        if month is not None:
+            filename += f"-{str(month).zfill(2)}"
+        if leadtime is not None:
+            filename += f"_lt{str(leadtime).zfill(2)}d"
+        filename += ".grib"
+        return directory / Path(filename)
+
+    def _get_query(
+        self,
+        area: Area,
+        version: int,
+        year: int,
+        month: int = None,
+        leadtime: int = None,
+    ) -> dict:
+        query = {
+            "variable": "total_precipitation",
+            'originating_centre': 'ecmwf',
+            'system': version,
+            "format": "grib",
+            self.dataset_variable_name: self.dataset,
+            "year": str(year),
+            "month": [str(x).zfill(2) for x in range(1, 13)]
+            if month is None
+            else str(month).zfill(2),
+            "leadtime_month":[str(x) for x in range(1,7)]
+            if leadtime is None
+            else str(leadtime),
+            "area": area.list_for_api(),
+        }
+        logger.debug(f"Query: {query}")
+        print(query)
+        return query
+
+    @staticmethod
+    def _read_in_monthly_mean_dataset(filepath_list: List[Path]):
+        """
+        Read in the dataset for each date and combine them
+        """
+
+        def _preprocess_monthly_mean_dataset(ds):
+            dsnew = ds.copy()
+            # step is in timedelta (in nanoseconds), where the timedelta is the end of the valid time of the forecast
+            # since the nanoseconds depends on the length of the month, convert this to the leadtime in months instead to be able to compare across months
+            #other option could be to convert it to the forecasted time, but gets difficult to concat all different publication dates afterwards
+
+            #todo: make this an input variable!
+            dsnew["step"] = range(1,7)
```

Comment:
Why 7? If it is a parameter of the dataset then you should get it from the dims, otherwise make it a constant